thoth/storages/graph/postgres.py

```diff
@@ -5477,6 +5477,34 @@ def sync_solver_result(self, document: dict, *, force: bool = False) -> None:
                                 marker_evaluation_result=dependency.get("marker_evaluation_result", True),
                             )
 
+            # Sync imoprted package from result:tree except for test module, all the namespaced calls
+            # modules starting with __ also excluded
+            package_name = document["result"]["tree"]["package_name"]
+            package_version = document["result"]["tree"]["package_version_requested"]
+            index_url = document["result"]["tree"]["index_url"]
+
+            _LOGGER.info(
+                "Syncing imported packages for package %r in version %r from %r found by solver %r",
+                package_name,
+                package_version,
+                index_url,
+                solver_info,
+            )
+
+            import_packages =[]
+            for package_call in document["result"]["tree"]["packages"]:
+                if package_call.count(".") == 0 and package_call[:4] != "test"
```

Comment:
No need to have any filtering here on names. You can directly create `ImportPackage` per each entry in `document["result"]["tree"]["packages"]`.

Reply:
resolved by 8a428279967604e0e603a2556370045264245cbd